sofria_mediaId.js

```diff
@@ -1,44 +1,144 @@
 
-// CLI: sofria-mediaId <USX FILESET DIRECTORY>
-// example: sofria-mediaId testdata/ENGESVN_ET-usx
-// result: create a corresponding sofria json fileset, eg:
-//  1. create testdata/ENGESVN_ET-json/ directory (note that it is adjacent to the USX FILESET DIRECTORY)
-//  2. populate it with single-chapter sofria json files (created from the input usx files)
+if (global.crypto !== 'object') {
+    global.crypto = {
+        getRandomValues: (array) => crypto.getRandomValues(array)
+    };
+}
 
-if (process.argv.length !== 5) {
-    throw new Error(`usage: sofria-mediaId <USX FILESET DIRECTORY>`);
+const progressBar = new cliProgress.SingleBar({}, cliProgress.Presets.shades_classic);
+
+let totalFilesToProcess = 0;
+let currentFileToProcess = 0;
+
+function increaseProgress() {
+    currentFileToProcess++;
+    progressBar.update(currentFileToProcess);
+
+    if (currentFileToProcess === totalFilesToProcess) {
+        progressBar.stop();
+    }
+}
+
+function initProgress(total) {
+    totalFilesToProcess = total;
+    progressBar.start(totalFilesToProcess);
+}
+
+function zeroComplete(num, places) {
+    return String(num).padStart(places, '0')
 }
 
+const offset = process.argv.length > 3 ? Number(process.argv[3]) : 0;
+const limit = process.argv.length > 4 ? Number(process.argv[4]) : Infinity;
+
 const pk = new Proskomma();
 
-/*
- FIXME: find all files in specified directory which have extension .usx. The expectation is that each usx file contains one book
- Loop over each usx book file and associate with fqPath
- */
-
-// lines 20-33 will be in the book loop, process a single usx book file. Used later, book_id would be of the form 040MAT
-const fqPath = xx
-
-const content = fse.readFileSync(fqPath).toString();
-const suffix = fqPath.split(".").reverse()[0];
-/// FIXME: change xxx and yyy to empty strings
-pk.importDocument({lang: "xxx", abbr: "yyy"}, suffix, content);
-// I assume that xxx and yyy are embedded in the json somewhere?
-const chaptersQuery = `{documents {cIndexes { chapter } } }`;
-const chaptersResult = pk.gqlQuerySync(chaptersQuery);
-const ret = {};
-for (const chapter of chaptersResult.data.documents[0].cIndexes.map(ci => ci.chapter)) {
+const fqPath = process.argv[2]
+const usxRootDirectory = fqPath.split("/").reverse()[0];
+
+async function getListFileFromDirectory(dirPath) {
+    try {
+        const filesUsx = await fse.readdir(dirPath);
+        const listFiles = [];
+        for (const fileUsx of filesUsx) {
+            const fileFormat = fileUsx.split(".");
+            const name = fileFormat[0] ? fileFormat[0] : "";
+            const suffix = fileFormat[1] ? fileFormat[1] : "";
+
+            listFiles.push({
+                fullpath: path.join(dirPath, fileUsx),
+                name,
+                suffix,
+                fullname: fileUsx,
+            });
+        }
+        return listFiles;
+    } catch (err) {
+        console.error('Error to read directory!', path, err);
```

Comment:
change message to Unable to read directory or filename
and add dirPath to the error message, in case the exception occurs on the readdir statement

Reply:
It has been updated.